cloud/Makefile

```diff
@@ -7,9 +7,20 @@ lint:
 	golangci-lint run --disable-all -E gofmt -E golint ./...
 	go vet ./...
 
+# Build and run tests.
+.PHONY: test
+ifeq ($(WHAT),)
+       TEST_DIR="./pkg/"
```

Comment:
I would use environment variables instead of the WHAT variable.
The execution of make could be:
make test WHAT=pkg (in your case)
TEST_DIR=pkg make test (in my case)
 
In the makefile this will be written as:
TEST_DIR ?= "./pkg/"

Reply:
@subpathdev , I have changed the test scripts . Now there is no requirement of WHAT variable.
# Running unit tests
```sh
$make test
$// verbose mode
$make test GOFLAGS=-v
$// select a package to run tests
$make test WHAT=pkg/controller/manager/ GOFLAGS=-v
$// select a test case to run
$make test GOFLAGS="-v" KUBE_TEST_ARGS="-run ^TestEventToMessage"
```
# Running integration tests
```sh
$make integration-test
$//verbose mode
$make integration-test GOFLAGS=-v
$// select a package to run tests
$make test-integration GOFLAGS=-v KUBE_TEST_ARGS="-run ^TestInvalidDeviceModel$"
$// select a folder to run tests
$make test-integration GOFLAGS=-v WHAT=./test/integration/crds/
```